tests/cases/compiler/circularReferenceInReturnType2.ts

```diff
@@ -0,0 +1,53 @@
+// @strict: true
```

Comment:
test case reduced from https://github.com/Thinkmill/graphql-ts

interestingly, this one also depends on a quite recent change from https://github.com/microsoft/TypeScript/pull/53246